### ISSUE TITLE
Design improvements - clear separtion between python extensions and core c++ lib

### ISSUE THREAD

**Issue by anubhav-cs** [anubhav-cs/LAPKT-private#9]
Currently, there is no clear separation between pybind11/boost-python extensions and the core C++ library which implements the STRIPS model and search algorithms, i.e. they are bundled together. This results in an extensive set of Python dependencies even when one simply wishes to use the core library without the wrappers.

The following changes are required to remove the dependency of the core LAPKT library from any extensions and wrappers written around it.  

- [x] Planner classes should not derive from the STRIPS Interface.  STRIPS Interface provides a translation service between PDDL, Python, and C++ implementation of the model, so they are unrelated to the core LAPKT libraries.
- [ ] A document explaining how to properly write pybind11 extensions around the core library. 
- [ ] A document explaining how to link third-party libraries, e.g. the inja, nholmann json, and ensure that we link the library against the correct node(library) in the dependency tree of LAPKT libraries.

In summary, we should avoid tainting our core C++ library which implements the STRIPS model and search algorithms with dependencies that are irrelevant to those functions.

**Comment by anubhav-cs** [comment]
created #10 #11 to track the documentation needs
